### PR TITLE
i18n for typst

### LINKED ISSUE #9555: The lang field does not alter key terms in typst
### Bug description

When the `lang` field is set in metadata, many key terms are not translated in typst as they are in other formats.

### Steps to reproduce

```yaml
---
title: Mi Título
format: typst
lang: es
abstract: Escribo algo.
---
```

In other formats, the abstract is translated to the language set in the `lang` field.

### Expected behavior

The abstract (and many other terms) should be in Spanish. In this case, the Abstract should be "Resumen"

### Actual behavior

Abstract is not translated. It is "Abstract" instead of "Resumen"

### Your environment

RStudio 2024.04.0+735 "Chocolate Cosmos" Release (a00d0e775dbc93e0d79a1bf474e3e8e8de677383, 2024-04-24) for windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2024.04.0+735 Chrome/120.0.6099.291 Electron/28.2.6 Safari/537.36, Quarto 1.5.31 (C:/Program Files/Quarto/bin/quarto.exe)

### Quarto check output

Quarto 1.5.31
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.13: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.5.31
      Path: C:\Program Files\Quarto\bin
      CodePage: 1252

[>] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)

[>] Checking LaTeX....................OK
      Using: TinyTex
      Path: C:\Users\renee\AppData\Roaming\TinyTeX\bin\windows\
      Version: 2024

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK
      Version: 3.11.3
      Path: C:/Users/renee/AppData/Local/Programs/Python/Python311/python.exe
      Jupyter: 5.3.1
      Kernels: python3

[>] Checking Jupyter engine render....OK

[>] Checking R installation...........OK
      Version: 4.4.0
      Path: C:/PROGRA~1/R/R-4.4.0
      LibPaths:
        - C:/Users/renee/AppData/Local/R/win-library/4.4
        - C:/Program Files/R/R-4.4.0/library
      knitr: 1.46.3
      rmarkdown: 2.26

[>] Checking Knitr engine render......OK

### LINKED ISSUE #9724: Support `abstract-title` for Typst format
I believe this could easily be supported like in HTML and some other formats by tweaking the template here

https://github.com/jgm/pandoc/blob/0f29d8195eeb526c4aa14d5f56818f905d5badb5/data/templates/template.typst#L62-L66

so that `Abstract` is not hard coded. 

Though it probably needs a Writer update to match behavior with Docx or HTML that supports this field

### PR DESCRIPTION
**Pull request by gordonwoodhull**
Closes #9555

Internationalization of
- Abstract title
- Figures
- Theorems etc

Loose ends on this draft
- [x] I don't know how to test a pandoc3 figure - where does this arise? I fixed this but didn't test it.
- [x] Similarly, what is a `PanelLayout` in Typst? I tried the HTML [Panel Layouts](https://quarto.org/docs/interactive/layout.html#panel-layout) `::: {layout = "[ [1,1] ]"}` but no surprise that didn't work.
   https://github.com/quarto-dev/quarto-cli/blob/301b2353dc93d07af83558a8defe48e46590f1c2/src/resources/filters/layout/typst.lua#L141
- [ ] Submit a PR to fix jgm/pandoc#9724 and see if this affects the change to the Typst template.

### CODE REVIEW

**Comment by cscheid** [comment]
> * Similarly, what is a `PanelLayout` in Typst? I tried the HTML [Panel Layouts](https://quarto.org/docs/interactive/layout.html#panel-layout) `::: {layout = "[ [1,1] ]"}` but no surprise that didn't work.

I believe you that it didn't work, but we do have support for it, in

https://github.com/quarto-dev/quarto-cli/blob/4b185d0c936a82a0888d6d31e288bffe889c4ede/src/resources/filters/layout/typst.lua#L86

**Comment by gordonwoodhull** [comment]
Thanks @cscheid! I was able to test pandoc3 figure with

```
![Hello]({{< placeholder >}})
```

and the subfloat case case for `PanelLayout` 

````qmd
::: {#fig-panel layout-ncol=2}

![Marcador]({{< placeholder 200 >}}){#fig-layout-a}

![Marcador]({{< placeholder 200 >}}){#fig-layout-b}

Panels

:::
````

IIUC the final case of `make_typst_figure()` in `layout/typst.lua` should be a `PanelLayout` without crossreferenceable content:

````qmd

::: {#fig-panel2 layout-ncol=2}

![Marcador]({{< placeholder 200 >}})

![Marcador]({{< placeholder 200 >}})

Panels

:::
````

But anything I put there, I get the inscrutable error

```
Block, list of Blocks, or compatible element expected, got Blocks
	while retrieving function argument content
	while retrieving arguments for function Div
```

(I thought that Blocks was a list of Blocks!)

**Comment by cscheid** [comment]
> (I thought that Blocks was a list of Blocks!)

This is a really bad error message from Pandoc, and it can come up in situations where a `Blocks` list has a non-Block element inside it. It is emitted by Pandoc in their Lua-to-Haskell marshalling code, and it's hard for us to guard against it in general :(

**Comment by gordonwoodhull** [comment]
Very helpful clue. Okay, I'm opening a new issue for that one as it doesn't seem like this example has worked right since 1.5.0 but it started crashing with subfloats.

I think this is complete except for looking at the future impact of the fix in Pandoc. That seems to require a different change to the template but I am not sure that I understand their i18n system yet.

**Comment by gordonwoodhull** [comment]
I've pushed the fix to #9854 to this branch, since the test here triggers that.

It turns out that the final localization is needed after all, because of the limitation that all typst figures are by definition cross-referenceable.

Ok to merge? When we get a fix upstream in Pandoc we can revert to their template.

**Comment by cscheid** [comment]
@gordonwoodhull I don't think we should have that subfigure situation - if we're going to have supplements and numbering in that case, then they'd need to be subfloats.

File: src/resources/filters/layout/typst.lua
Comment (gordonwoodhull) [inline src/resources/filters/layout/typst.lua:1]: commit message is incorrect, this affects the parent title of a panel with non-cross-referenceable figures (that are nevertheless rendered as cross-referenced, with counter continuing in them)

**Comment by gordonwoodhull** [comment]
Completely agree, but I think it's outside the scope of this PR. I think we have a small constellation of issues around figures without ids in Typst.

Could we merge this and I'll open another issue with the numbering problem?

**Comment by gordonwoodhull** [comment]
Continued in #9857

**Comment by gordonwoodhull** [comment]
Squashed and merged to main in d40b8ec70, closing this.

**Pull request closed without merging**